packages/design-system/src/components/Dropdown/DropdownMenu.tsx

```diff
@@ -60,8 +60,13 @@ export const DropdownMenu = ({
   React.useEffect(() => {
     const handleFocus = (event: FocusEvent) => {
       const target = event.target as HTMLElement;
-
-      if (!focusManager.dropdown?.current?.contains(target)) {
+      if (
+        !focusManager.dropdown?.current?.contains(target) &&
+        // Do not close the dropdown if the event target is modal content,
+        // otherwise dropdowns inside of modals are broken.
+        // This allows the event to propagate down to the correct input component.
+        !target.className.includes("ReactModal__Content")
```

Comment:
fascinating ... while I am certainly not opposed to hackery, what I don't like about this solution is that clicking on `.ReactModal__Content` is in theory a perfectly valid case for closing the menu and I wouldn't want to break that.

I will confess that I have never really understood how the focus management works in this component, and possibly now less than ever lol ... I think what I am confused by here is that ..... events propagate up, not down? Why would we be getting an event from this element before handling the one inside the menu? What is special about `.ReactModal__Content` that makes the order of events different from any of the other elements that lie between `document` and `target`? I suspect maybe it has to do with the fact that, in the Workflows use case that prompted this, we are actually listening for checkbox `change` events, which fire after the `click` events that the `Dropdown` is expecting to handle with the `focusManager`, and the fact that this ever worked at all was just, like ........ an accident?

Reply:
> clicking on .ReactModal__Content is in theory a perfectly valid case for closing the menu and I wouldn't want to break that.

It does not break the use case for clicking inside the modal, but outside of the dropdown to close the dropdown. You can see that working [here](https://recidiviz-dashboard-stag-e1108--rh-panel-z0gathid.web.app/workflows). This is because the event target is outside of the dropdown menu so  line 64 does its job and closes the dropdown.

> What is special about .ReactModal__Content that makes the order of events different from any of the other elements that lie between document and target? Why would we be getting an event from this element before handling the one inside the menu?

I'm not really sure. When logging the `target` on line 63, if I revert the if statement to what it was previously, this is the sequence of target elements:
<img width="1761" alt="Screen Shot 2022-10-07 at 10 16 46 AM" src="https://user-images.githubusercontent.com/5872651/194613904-5d73c0a9-8a44-4897-b8fa-fb88229b14ad.png">

When I add the line to ignore the modal content target, this is the sequence:
<img width="1775" alt="Screen Shot 2022-10-07 at 10 12 56 AM" src="https://user-images.githubusercontent.com/5872651/194613984-4cfcbd62-5b3c-4f75-8af7-619fdb0e7323.png">

You can see that it goes from the modal content, to the label, to the input, so it does seem to be moving down?

Reply:
it does break it ... in that example if you click within the modal's padding area (which is like 16px wide) the menu doesn't close

Reply:
I wonder if it would be possible to work around this in Workflows by listening to click events on the checkboxes instead of change events?

Reply:
> I wonder if it would be possible to work around this in Workflows by listening to click events on the checkboxes instead of change events?

PS that did not work. The click event was never reaching the input.

Reply:
I didn't think we were in the capture phase here because doesn't that have to be explicitly passed as an option to `addEventListener`? I still don't really understand what's going on here though so ... whatever

Would switching to `focus` mean that one could not close the menu by focusing elsewhere in the UI (i.e. "tabbing out" of it to another button or something)? I would guess that is why we used `focusin` in the first place but probably only @ohaibbq knows for sure. Anyway I don't think that's a dealbreaker, I don't see any specific guidance on that in [the WAI-ARIA design pattern](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/)

Reply:
Oh you're right. With it changed to `focus` tabbing out of the dropdown does not close the dropdown. "Esc" closes the dropdown but tabbing out no longer works.

Reply:
`document.addEventListener("focusin", handleFocus, false);` - the last `false` argument is to capture events on bubble instead of capture, fixes the issue and still closes the dropdown on tabbing out. I'll push up a commit with that.